src/v/cloud_topics/level_one/metastore/state_update.h

```diff
@@ -48,10 +48,13 @@ struct new_object
     };
 
     friend bool operator==(const new_object&, const new_object&) = default;
-    auto serde_fields() { return std::tie(oid, extent_metas); }
+    auto serde_fields() {
+        return std::tie(oid, footer_pos, object_size, extent_metas);
+    }
 
     object_id oid;
     size_t footer_pos;
+    size_t object_size;
```

Comment:
I thought about if it makes sense to validate the footer_pos < object_size, but that feels pretty defensive TBH.